docs-src/http-routing.md

```diff
@@ -0,0 +1,63 @@
+# HTTP routing
+
+HTTPRoute resources allow you to match on HTTP traffic and direct it Kubernetes
+backends. This guide shows how the HTTPRoute matches on host, header, and path
+fields and forwards traffic to different Kubernetes Services.
+
+The following diagram describes a required traffic flow across three different
+Services:
+
+- Traffic to `foo.example.com/login` is forwarded to `foo-svc` 
+- Traffic to `bar.example.com/*` with a `Env: canary` header is forwarded 
+to `bar-svc-canary`
+- Traffic to `bar.example.com/*` without the header is forwarded to `bar-svc` 
+- Any remaining non-matching traffic is responded to with an HTTP 404 message
+
+![HTTP Routinng](./images/http-routing.png)
+
+The dotted lines show the Gateway resources deployed to configure this routing
+behavior. There are two HTTPRoute resources that create routing rules on the
+same `prod-web` Gateway. This illustrates how [more than one Route can bind to a
+Gateway](api-overview.md/#gateway) which allows Routes to merge on a Gateway as
+long as they don't conflict. 
+
+The following `prod-web` Gateway is defined from the `acme-lb` GatewayClass.
+`prod-web` listens for HTTP traffic on port 80 and will bind to all Routes in
+the `default` Namespace that have the matching labels  `gateway: prod-web-gw`.
+Route labels allow the Route to determine which Gateways recieve their routing
+rules.  
+
+```yaml 
+{% include 'http-routing/gateway.yaml' %}  
+```
+
+An HTTPRoute can match against a [single set of
+hostname](https://kubernetes-sigs.github.io/service-apis/spec/#networking.x-k8s.io/v1alpha1.HTTPRouteSpec).
+These hostnames are matched before any other matching within the HTTPRoute takes
+place. Since `foo.example.com` and `bar.example.com` are separate hosts with
+different routing requirements, each is deployed as its own HTTPRoute -
+`foo-route` and `bar-route`.
+
+The following `foo-route` will match any traffic for `foo.example.com` and apply
+its routing rules to forward the traffic to the correct backend. Since there is
+only one match specified, only `foo.example.com/login*` traffic will be
```

Comment:
It looks like we have left the matching rather ambiguous here, but in the Ingress spec, a prefix match for `login` would match requests to `/login` or `/login/foo` but not `/loginfoo`. I think that's likely worth carrying over to this spec, or at least assuming for now.

Reply:
changed this to /login/*